app/models/custom_attributes/custom_attribute_value.rb

```diff
@@ -20,6 +20,22 @@ def value_or_display_value
       def value_present?
         date_time_value.present? || string_value.present? || double_value.present? || integer_value.present? || !boolean_value.nil? || custom_attribute_option_values.any?
       end
+
+      def no_of_days_in_year
+        Date.leap?(DateTime.now.year) ? 366 : 365
+      end
+
+      def date_for_next_alert
+        if date_time_value > Date.today
+          date_time_value
+        else
+          time_cycle    = custom_attribute_defn.repeat_cycle.downcase
+          divisor       = { day: 1, week: 7, month: Date.today.end_of_month.day, year: no_of_days_in_year }[time_cycle.to_sym]
```

Comment:
there is no need to have this logic as earlier mentioned this
use like
```
date_time_value + custom_attribute_defn.repeat_duration.send(custom_attribute_defn.repeat_cycle)
```
but make sure that these value stored in db are correct and this can be done by model validation